crates/interpreter/src/valid.rs

```diff
@@ -25,23 +25,47 @@ use crate::toctou::*;
 use crate::util::*;
 use crate::*;
 
+pub trait ValidMode {}
```

Comment:
We can have `trait ValidMode: Default {}` for convenience, then we can just write `ValidMode` instead of `ValidMode + Default`.